components/CreateOrUpdateRecipeForm.js

```diff
@@ -0,0 +1,212 @@
+import styled from "styled-components";
+import { useRouter } from "next/router";
+import Link from "next/link";
+
+export default function CreateOrUpdateRecipeForm({
+  errorMessage,
+  inputValidation,
+  recipeDetails,
+  onUpdateRecipe,
+  onHandleSubmit,
+  isUpdate,
+}) {
+  const router = useRouter();
+
+  return (
+    <>
+      <h1>{isUpdate ? "Update Recipe" : "Create new recipe"}</h1>
+
+      {errorMessage ? (
+        <StyledErrorMessageContainer>
+          <StyledErrorMessage>{errorMessage}</StyledErrorMessage>
+        </StyledErrorMessageContainer>
+      ) : (
+        ""
+      )}
+      <StyledForm onSubmit={isUpdate ? onUpdateRecipe : onHandleSubmit}>
```

Comment:
```suggestion
      <StyledForm onSubmit={onSubmit}>
```
Here you wouldn't have to handle the onSubmit conditionally, when you only use one `onSubmit` prop for your form component :)